plotter/scatterColor_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"testing"
+
+	"gonum.org/v1/plot"
+	"gonum.org/v1/plot/internal/cmpimg"
+	"gonum.org/v1/plot/palette/moreland"
+	"gonum.org/v1/plot/vg"
+	"gonum.org/v1/plot/vg/draw"
+	"gonum.org/v1/plot/vg/vgimg"
+)
+
+// ExampleScatter_color draws some scatter points.
+// Each point is plotted with a different color depending on
+// some external criteria.
+func ExampleScatter_color() {
+	rnd := rand.New(rand.NewSource(1))
+
+	// randomPoints returns some random x, y points
+	// with some interesting kind of trend.
+	randomPoints := func(n int) XYs {
+		pts := make(XYs, n)
+		for i := range pts {
+			if i == 0 {
+				pts[i].X = rnd.Float64()
+			} else {
+				pts[i].X = pts[i-1].X + rnd.Float64()
+			}
+			pts[i].Y = pts[i].X + 10*rnd.Float64()
+		}
+		return pts
+	}
+
+	n := 15
+	scatterDataNew := randomPoints(n)
+
+	p, err := plot.New()
+	if err != nil {
+		log.Panic(err)
+	}
+	p.Title.Text = "Points Example Color"
+	p.X.Label.Text = "X"
+	p.Y.Label.Text = "Y"
+	p.Add(NewGrid())
+
+	sc, err := NewScatter(scatterDataNew)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	colors := moreland.Kindlmann() // Initialize a color map.
+	colors.SetMax(255)
+	colors.SetMin(0)
+
+	// Variable z depends on i, and used for setting color for each individual point of the plot.
+	z := []float64{0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 110, 120, 130, 140, 150}
+
+	//Specify style for individual points.
+	sc.GlyphStyleFunc = func(i int) draw.GlyphStyle {
+		c, err := colors.At(z[i])
+		if err != nil {
+			log.Panic(err)
+		}
+		return draw.GlyphStyle{Color: c, Radius: vg.Points(3), Shape: draw.CircleGlyph{}}
+	}
+	p.Add(sc)
+
+	// Create a legend.
+	thumbs := PaletteThumbnailers(colors.Palette(n))
```

Comment:
I think this would work better with a palette.ColorMap instead of a palette.Palette. I think I put some example code for that somewhere in this thread.

Reply:
I have actually replied about this issue - here is the problem I am facing - colors are taking the Y axis values which is wrong, I am sure. 
![scattercolor](https://user-images.githubusercontent.com/18654639/32406575-1cfcc92c-c18c-11e7-9a9b-744017705408.png)